drivers/timer/stm32_lptim_timer.c

```diff
@@ -429,6 +429,23 @@ static int sys_clock_driver_init(void)
 	}
 #endif
 
+#if DT_PROP(DT_DRV_INST(0), st_counter_value)
+	/* Calculate the corresponding prescaler */
+	for (lptim_clock_presc = 1; lptim_clock_presc <= 32 ; lptim_clock_presc++) {
+		if (lptim_clock_presc > (uint32_t)DT_PROP(DT_DRV_INST(0), st_counter_value)) {
+			break;
+		}
+	}
```

Comment:
```suggestion
	lptim_clock_presc = MAX(1, (uint32_t)DT_PROP(DT_DRV_INST(0), st_counter_value));
	lptim_clock_presc = MIN(32, lptim_clock_presc);
```
I am not sure whether this faster and more readable, just a suggestion.

Reply:
_more readable_ for sure

Reply:
file:///local/home/frq09495/Pictures/Screenshots/Screenshot%20from%202024-02-19%2010-55-57.png

Reply:
Ok, I was under the impression that the loop could return a value between 1 and 32 and not only power of 2.